runtime/convertValues.go

```diff
@@ -64,123 +50,150 @@ func exportValueWithInterpreter(
 	value interpreter.Value,
 	inter *interpreter.Interpreter,
 	results exportResults,
-) cadence.Value {
-
-	if result, ok := results[value]; ok {
-		return result
+) (
+	cadence.Value,
+	error,
+) {
+	// Break recursion through ephemeral references
+	if referenceValue, ok := value.(*interpreter.EphemeralReferenceValue); ok {
+		if _, ok := results[referenceValue]; ok {
+			return nil, nil
+		}
+		defer delete(results, referenceValue)
+		results[referenceValue] = struct{}{}
```

Comment:
Can this be moved to the switch-case below, under `case *interpreter.EphemeralReferenceValue`?

Reply:
Good observation! 👍 Refactored in 4e80241a74378dabb442a4259bb53d6f0c7cc050